### PR TITLE
Make in-memory cross-project references the only option (by removing the switch)

### PR DESCRIPTION
**Pull request by vzarytovskii**

### CODE REVIEW

File: vsintegration/src/FSharp.UIResources/LanguageServicePerformanceOptionControl.xaml [hunk 1]
```diff
@@ -19,10 +19,6 @@
             <StackPanel>
                 <GroupBox Header="{x:Static local:Strings.Project_Performance}">
                     <StackPanel>
-                        <CheckBox x:Name="enableInMemoryCrossProjectReferences"
-                                  IsChecked="{Binding EnableInMemoryCrossProjectReferences}"
-                                  Content="{x:Static local:Strings.Enable_in_memory_cross_project_references}"
-                                  ToolTip="{x:Static local:Strings.Tooltip_in_memory_cross_project_references}"/>
```
Comment (0101) [inline vsintegration/src/FSharp.UIResources/LanguageServicePerformanceOptionControl.xaml:25]: We should probably also remove these strings from `Strings.resx` file.

Comment (vzarytovskii) [reply vsintegration/src/FSharp.UIResources/LanguageServicePerformanceOptionControl.xaml:25]: Oh, I must've missed, was wondering why translation baseline wasn't updated

**Review by 0101 — APPROVED** [review]

**Comment by vzarytovskii** [comment]
> I am all up for removing whatever code - but what's the exact motivation here?

It was on by default for a long time, no reason of keeping it really.

**Comment by majocha** [comment]
Yes, not wired up to the checker and probably only confusing new users.

**Review by KevinRansom — APPROVED** [review]

**Comment by vzarytovskii** [comment]
> Yes, not wired up to the checker and probably only confusing new users.

You mean a different one, I believe.

**Comment by psfinaki** [comment]
I see. Looking at telemetry, nobody ever turned it off (except you Vlad :D )

**Comment by vzarytovskii** [comment]
> except you Vlad

You know what's the awkward part? I DON'T REMEMBER DOING IT!

**Comment by TIHan** [comment]
Oo this is good!

**Comment by vzarytovskii** [comment]
Spoken to @charlesroddie, and will roll it back for time being, since it makes a difference in some project graphs. 

Until we have a better solution.

**Comment by auduchinok** [comment]
> Spoken to @charlesroddie, and will roll it back for time being, since it makes a difference in some project graphs.

@vzarytovskii Could you please tell more about these differences?

**Comment by vzarytovskii** [comment]
> > Spoken to @charlesroddie, and will roll it back for time being, since it makes a difference in some project graphs.
> 
> @vzarytovskii Could you please tell more about these differences?

Apparently, on solution @charlesroddie have, it causes a significant slowdown.

**Comment by charlesroddie** [comment]
> > Spoken to @charlesroddie, and will roll it back for time being, since it makes a difference in some project graphs.
> 
> @vzarytovskii Could you please tell more about these differences?

I was worried that eventually someone would turn off this switch, as it became at some point not the default but I was relying on it for acceptable and predictable performance for many years. That happened, but I am very grateful and impressed with the support and speed of reversion!

When I updated Visual Studio I found that it took 3mins instead of 5s for intellisense to show up on some projects, because of the removal of this mode.

We have 197395 F# LOC in the solution. I suppose that's a lot. A small majority is specification of data as a DSL in F# code which is how we have so many lines.

It performs well because it's split up into many projects, so when editing files, the compiler only need to work with a few lines of F# code. Building takes a while (1.2mins not 5) and I'm looking forward to ditching mono soon and being able to use F# reference assembly support which will make most of our builds almost instant.

I think many people are missing out by not being aware of the "disable in memory cross-project references" mode.

**Pull request merged**